hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnection.java

```diff
@@ -269,21 +271,32 @@ public String getCloseReason() {
     }
 
     @SuppressFBWarnings(value = "VO_VOLATILE_INCREMENT", justification = "incremented in single thread")
-    void heartBeatingFailed() {
-        heartBeating = false;
+    void heartbeatFailed() {
+        isHeartBeating = false;
     }
 
-    void heartBeatingSucceed() {
-        heartBeating = true;
-        lastHeartbeatMillis = System.currentTimeMillis();
+    void heartbeatResumed() {
+        isHeartBeating = true;
     }
 
-    public long getLastHeartbeatMillis() {
-        return lastHeartbeatMillis;
+    void heartbeatReceived() {
```

Comment:
this name doesn't explain if it is an 'getter' or 'do something'. Probably using 'onHeartBeatFailed' would be more descriptive.